Wrappers/d3d8/IDirect3DDevice8.cpp

```diff
@@ -805,19 +866,198 @@ HRESULT m_IDirect3DDevice8::DrawPrimitive(D3DPRIMITIVETYPE PrimitiveType, UINT S
 		}
 	}
 
+	DWORD stencilPass = 0;
+	ProxyInterface->GetRenderState(D3DRS_STENCILPASS, &stencilPass);
+
+	if (EnableXboxShadows() && stencilPass == D3DSTENCILOP_INCR && !shadowVolumeFlag)
+	{
+		// We're drawing shadow volumes, toggle flag on
+		shadowVolumeFlag = true;
+
+		IDirect3DSurface8 *backbufferColor, *backbufferDepthStencil = nullptr;
+
+		// Backup backbuffer and depth/stencil buffer
+		if (SUCCEEDED(ProxyInterface->GetRenderTarget(&backbufferColor)) && backbufferColor)
+		{
+			backbufferColor->Release();
+		}
+		if (SUCCEEDED(ProxyInterface->GetDepthStencilSurface(&backbufferDepthStencil)) && backbufferDepthStencil)
+		{
+			backbufferDepthStencil->Release();
+		}
+
+		// Create texture for James' silhouette if it doesn't exist
+		if (!silhouetteTexture)
+		{
+			if (SUCCEEDED(ProxyInterface->CreateTexture(BufferWidth, BufferHeight, 1, D3DUSAGE_RENDERTARGET, D3DFMT_A8R8G8B8, D3DPOOL_DEFAULT, &silhouetteTexture)) && silhouetteTexture)
+			{
+				silhouetteTexture->GetSurfaceLevel(0, &silhouetteSurface);
+			}
+		}
+
+		// Temporarily swap to new color buffer (keep original depth/stencil)
+		ProxyInterface->SetRenderTarget(silhouetteSurface, backbufferDepthStencil);
```

Comment:
`silhouetteSurface` is used for the first time here, but it is never validated before it is used.  If `GetSurfaceLevel` fails for whatever reason the game will crash here.